File: examples/java-maven-project/src/main/java/org/ton/sdk/example/GetTransactionExample.java
package org.ton.sdk.example;

import org.ton.sdk.blockchain.address.AddressStd;
import org.ton.sdk.blockchain.currency.Coins;
import org.ton.sdk.toncenter.client.TonCenterV3Client;
import org.ton.sdk.toncenter.model.TonCenterTransaction;
import org.ton.sdk.toncenter.model.TonCenterTransactionsRequestBuilder;
import org.ton.sdk.toncenter.model.TonCenterTransactionsResponse;

import java.math.BigInteger;
import java.util.concurrent.ExecutionException;
import java.util.concurrent.Future;

public class GetTransactionExample {
    public static void main(String[] args) throws ExecutionException, InterruptedException {
        // create TON Center v3 http client
        TonCenterV3Client client = TonCenterV3Client.create();

        // Get 10 last transactions for account
        Future<TonCenterTransactionsResponse> response = client.transactionsAsync(
                new TonCenterTransactionsRequestBuilder()
                        .address(AddressStd.parse("UQAKtVj024T9MfYaJzU1xnDAkf_GGbHNu-V2mgvyjTuP6uYH"))
                        .limit(10)
        );

        // Print transactions info and balance after transaction
        for (TonCenterTransaction transaction : response.get().transactions()) {
            Coins balance = transaction.accountStateAfter().balance();

            BigInteger value = BigInteger.ZERO;
            if (balance != null) {
                value = balance.value();
            }
            System.out.println("hash="+transaction.hash()+" lt="+transaction.lt()+" balance="+value);
        }
    }
}

File: examples/kotlin-gradle-project/src/jvmMain/kotlin/GetTransactionExample.kt
@file:Suppress("OPT_IN_USAGE")

package org.ton.sdk.example

import kotlinx.coroutines.Deferred
import kotlinx.coroutines.GlobalScope
import kotlinx.coroutines.async
import kotlinx.coroutines.runBlocking
import org.ton.sdk.bigint.toBigInt
import org.ton.sdk.blockchain.address.AddressStd
import org.ton.sdk.toncenter.client.TonCenterV3Client
import org.ton.sdk.toncenter.client.transactions
import org.ton.sdk.toncenter.model.TonCenterTransactionsResponse

object GetTransactionExample {
    @JvmStatic
    fun main(args: Array<String>) = runBlocking {
        // create TON Center v3 http client
        val client: TonCenterV3Client = TonCenterV3Client.create()

        // Get 10 last transactions for account
        val response: Deferred<TonCenterTransactionsResponse> = GlobalScope.async {
            client.transactions {
                account += AddressStd.parse("UQAKtVj024T9MfYaJzU1xnDAkf_GGbHNu-V2mgvyjTuP6uYH")
                limit = 10
            }
        }

        // Print transactions info and balance after transaction
        response.await().transactions.forEach { transaction ->
            val balance = transaction.accountStateAfter.balance
            val value = balance?.value ?: 0.toBigInt()
            println("hash=${transaction.hash} lt=${transaction.lt} balance=$value")
        }
    }
}
